core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -129,7 +130,7 @@ public void addNode(final @Nonnull Node node) throws IOException {
         if (node != nodes.get(node.getNodeName())) {
             // TODO we should not need to lock the queue for adding nodes but until we have a way to update the
             // computer list for just the new node
-            Queue.withLock(new Runnable() {
+            withLock(new Runnable() {
```

Comment:
non-reentrant behaviour from two Future.get() methods chasing their tails...

I'm rejecting this whole change as not worth the effort